src/testWithSpringBoot_2_4/java/org/openrewrite/java/spring/boot2/DatabaseComponentAndBeanInitializationOrderingTest.java

```diff
@@ -380,32 +552,102 @@ public MagicDataSourceInitializer magicDataSourceInitializer() {
     void componentDoesDdl() {
         //language=java
         rewriteRun(
-          java(
-            """
-              import javax.sql.DataSource;
-              import org.springframework.stereotype.Component;
-
-              @Component
-              public class MyDbInitializerComponent {
-
-                  public void initSchema(DataSource ds) {
-                  }
-              }
-              """,
-            """
-              import javax.sql.DataSource;
-
-              import org.springframework.boot.sql.init.dependency.DependsOnDatabaseInitialization;
-              import org.springframework.stereotype.Component;
-
-              @Component
-              @DependsOnDatabaseInitialization
-              public class MyDbInitializerComponent {
-
-                  public void initSchema(DataSource ds) {
-                  }
-              }
+          mavenProject("project-maven",
+            //language=xml
+            pomXml(
+              """
+                <project>
+                    <groupId>com.example</groupId>
+                    <artifactId>foo</artifactId>
+                    <version>1.0.0</version>
+                    <dependencies>
+                      <dependency>
+                        <groupId>org.springframework.boot</groupId>
+                        <artifactId>spring-boot</artifactId>
+                        <version>2.5.1</version>
+                      </dependency>
+                    </dependencies>
+                </project>
+                """
```

Comment:
Should we extract a constant here to make it clear which tests are using the same, and have the other one stand out more?